### DISCUSSION TITLE
Lesson 7, Part 5 - testing on VScode..."Member "i_owner" not found or not visible after argument-dependent lookup in contract FundMe."

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#495] [General]
I am working with FundMeTest.t.sol and I tried to add the test that i_owner is the same as address(this) and I got the error "Member "i_owner" not found or not visible after argument-dependent lookup in contract FundMe."  when I ran forge test. A screen cap of my VScode is below. 

I had a bunch of errors when I tried to run forge test before adding the testOwnerIsMsgSender function as well and I was able to resolve them with workarounds from Googling, but now that I am hitting another error, I am thinking I should ask if there is something I am overlooking with all of this. One of the errors I had to resolve before was that the constructor function in FundMe.sol takes one parameter but when I created "new FundMe()" I wasn't passing that parameter (which seemed to work for Patrick but not me...I had to add the address for priceFeed into the parentheses). I think this is all related.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/73b923db-7aa7-4b8a-b12a-e3a2a0e1c838)

**Comment by theirrationalone** [comment]
Go to your FundMe.sol and cross check that statement 👇 below is same as it is here.

```solidity
address public immutable i_owner;
```

private and internal functions aren't visible to outside the contract. Do you remember what Patrick said... 

A getter function is automatically created when we declare & define a Public state variable. That's why it's
```
i_owner(); // <- (a function call)
```
👆Just because of public. In your case it might be... 👇

```solidity
address private immutable i_owner;
```

Try to make that variable public or You can also try to make a getter function of i_owner i.e. 👇,
```solidity
function getOwner() public view returns (address) {
    return i_owner;
}
```

Thank You 🙏🙏🙏

**Reply by 0xwtfwtf** [reply]
thank you

**Reply by vectordotsats** [reply]
Thanks @theirrationalone , i really appreciate.

**Reply by Benio501st** [reply]
thank you
